hs_core/views/__init__.py

```diff
@@ -318,6 +347,21 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
         _set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=True)
     elif t == 'make_private' or t == 'make_not_discoverable':
         _set_resource_sharing_status(request, user, res, flag_to_set='public', flag_value=False)
+
+       # reverse lookup: find all associated collection resources
+        associated_collection_metadata_obj_list = list(res.collection_set.all())
```

Comment:
@pkdash This piece of code and the above one you mentioned are to implement this feature:
CollectionA contains public ResB, so user made Collection A public.
If user change ResB to private (note this happens on ResB's landing page, not on Collection A's), we need to downgrade all public collections that contain ResB to private (in this case, CollectionA). Another related but different scenario, if user delete ResB (this also happens on ResB's landing page), then CollectionA holds 0 resources which does meet public criteria anymore, so we need to downgrade CollectionA to private.

This custom logic is exclusive for Collection, but it is triggered on other res landing pages. so I put this logic here. If you think there is a better place to put it, please guide me to move it there. thanks